script.js

```diff
@@ -1,4 +1,184 @@
+      return `Ok, your hand is ${pOneResult}. <br><br>It is now Player 2's turn.`;
+    } else {
+      return `Please select a valid number.<br><br>${myOutputValue}`;
+    }
+  } // Player 2 choose order of die
+  else if (turnOrOrder == 3 && whichDiceOrder == 1) {
+    if (input == 1) {
+      console.log("p2 1");
+      turnOrOrder = 4;
+      whichDiceOrder = 0;
+      pTwoResult = "" + playerThrow[0] + playerThrow[1];
+      myOutputValue = `Ok, your hand is ${pTwoResult}.`;
+      return myOutputValue;
+    } else if (input == 2) {
+      console.log("p2 2");
+      turnOrOrder = 4;
+      whichDiceOrder = 0;
+      pTwoResult = "" + playerThrow[1] + playerThrow[0];
+      myOutputValue = `Ok, your hand is ${pTwoResult}`;
+      return myOutputValue;
+    } else {
+      return `Please select a valid number.<br><br>${myOutputValue}`;
+    }
+  }
+
+  // Evaluate results for winner
+  else if (turnOrOrder == 4 && whichDiceOrder == 0) {
+    turnOrOrder = 0;
+    pOneStats = Math.floor(pOneStats) + Math.floor(pOneResult);
+    pTwoStats = Math.floor(pTwoStats) + Math.floor(pTwoResult);
+    gameStats();
+    //P2 results bigger
+    // add in game mode condition
+    if (
+      (pOneResult > pTwoResult && gameMode == 1) ||
+      (pOneResult < pTwoResult && gameMode == 2)
+    ) {
+      myOutputValue =
+        myOutputValue +
+        ` but Player 1's hand is ${pOneResult}. You lost :(<br><br>The game will now restart!` +
+        leaderboard;
+      return myOutputValue;
+      //P2 results smaller
+    } else if (
+      (pTwoResult > pOneResult && gameMode == 1) ||
+      (pTwoResult < pOneResult && gameMode == 2)
+    ) {
+      myOutputValue =
+        myOutputValue +
+        ` and Player 1's hand is ${pOneResult}. You won! :)<br><br>The game will now restart!` +
+        leaderboard;
+      return myOutputValue;
+      //P2 results smame
+    } else if (pTwoResult == pOneResult) {
+      myOutputValue =
+        myOutputValue +
+        ` and Player 1's hand is ${pOneResult}. It's a draw! :)<br><br>The game will now restart!` +
+        leaderboard;
+      return myOutputValue;
+    }
+  } else if (turnOrOrder == -1 && input == 0) {
+    return `Welcome players, lets play a game of Beat That! <br><br>Rules:<br>Each player will take turn to roll 2 dice and choose the order they want to arrange the dice. <br><br> In normal mode, the player with the bigger hand wins while in reversed mode, the player with the smaller hand wins!<br><br> Please indicate whether you would like to play normal or reversed mode:<br>1) Normal Mode<br>2) Reversed Mode`;
+  } else if (turnOrOrder == -1 && (input == 1 || input == 2)) {
+    turnOrOrder += 1;
+    if (input == 1) {
+      gameMode = 1;
+      return `All right, lets start with the Normal game mode! Player 1 please click to roll.`;
+    } else {
+      gameMode = 2;
+      return `All right, lets start with the Reversed game mode! Player 1 please click to roll.`;
+    }
```

Comment:
I would put this at the top of the function, always cater for the special or rarer scenarios first and return early, then move on to the rest of your code. This also makes sense intuitively because your code actually starts out with `turnOrOrder` being -1 and comes in here first. 

Also, try seeing how if you can put line 150 into the output box immediately on page load by editing the html so that the user sees the message straight away when he loads the game. Currently when I load your game, I am greeted with an empty text input and output box, and it is not immediately clear what I have to do. Always think about user experience when you are designing something!

Also for the case in line 151 onwards, some input validation is probably needed because if I enter something that is not 1 or 2, I get back 'undefined' in the output box.